### ISSUE TITLE
[upload] workdir cleanup fails: PermissionError: Access is denied

### ISSUE THREAD

**Issue by eachimei** [devpi/devpi#913]
devpi-client 6.0
Windows 10
Python 3.8

The upload flow (devpi.upload:main) is executed inside `with hub.workdir() as uploadbase:`:
```python
    with hub.workdir() as uploadbase:
        exported = checkout.export(uploadbase)
```
and therefore it will try to remove (`rmtree`) workdir at `__exit__` (devpi.main.Hub:workdir): 
```python
    @contextmanager
    def workdir(self, prefix='devpi-'):
        workdir = py.path.local(
            mkdtemp(prefix=prefix))

        self.info("using workdir", workdir)
        try:
            yield workdir
        finally:
            rmtree(workdir.strpath)
```

However, `checkout.export()` that runs inside this context copies also the **.git** folder, and the object files inside it are **read-only**. 
As a result, the `rmtree` command fails.

```
Traceback (most recent call last):
  File "C:\Python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "<...>\.tox\release_new\lib\site-packages\devpi\__main__.py", line 3, in <module>
    main.main()
  File "<...>\.tox\release_new\lib\site-packages\devpi\main.py", line 40, in main
    return method(hub, hub.args)
  File "<...>\.tox\release_new\lib\site-packages\devpi\upload.py", line 50, in main
    uploader.do_upload_paths(archives)
  File "C:\Python38\lib\contextlib.py", line 120, in __exit__
    next(self.gen)
  File "<...>\.tox\release_new\lib\site-packages\devpi\main.py", line 239, in workdir
    rmtree(workdir.strpath)
  File "C:\Python38\lib\shutil.py", line 740, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "C:\Python38\lib\shutil.py", line 613, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\Python38\lib\shutil.py", line 613, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\Python38\lib\shutil.py", line 613, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  [Previous line repeated 1 more time]
  File "C:\Python38\lib\shutil.py", line 618, in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
  File "C:\Python38\lib\shutil.py", line 616, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 5] Access is denied: 'C:\\Users\\<...>\\AppData\\Local\\Temp\\devpi-pi5ir4lb\\<...>\\.git\\objects\\00\\bd0e983e668d4378933db5affa961fe527f952'
ERROR: InvocationError for command '<...>\.tox\release_new\Scripts\python.EXE' -m devpi upload --only-docs --debug (exited with code 1)
```

**Comment by fschulze** [comment]
Would the ``--setupdir-only`` or ``--no-vcs`` options work for you? Otherwise we would need to change the export to either change the permissions on copy, or maybe use newer git features like [``git worktree``](https://git-scm.com/docs/git-worktree).

**Comment by eachimei** [comment]
Thanks for the prompt response!

I am using `setuptools_scm` which is integrated in `pyproject.toml` according to PEP-517. Therefore, I am a bit concerned regarding `--setupdir-only` (I am afraid that something will break).
Also, I would like to keep using the export (avoid `--no-vcs`), it sounds like a good practice and a good safety-measure.

How about making the `rmtree` a bit "stronger", at least in that context? Please see: https://stackoverflow.com/a/1889686

**Comment by eachimei** [comment]
> How about making the `rmtree` a bit "stronger", at least in that context? Please see: https://stackoverflow.com/a/1889686

Tested with this change (`onerror` handler for `rmtree`) - works properly

**Comment by eachimei** [comment]
Fixed in PR #916

**Comment by henryborchers** [comment]
Super excited to see this being closed. I can't wait to try the new client when it gets released. This issue was the only thing holding me back from using version 6

**Comment by fschulze** [comment]
I have to clarify: This is a shot in the dark workaround. I have no idea if it really fixes it, as I can't reproduce it and I'm not sure what causes it.

If you can, please try the dev-release:
https://m.devpi.net/fschulze/dev/devpi-client/6.0.3.dev0
https://m.devpi.net/fschulze/dev/+f/546/62fdd859248fb/devpi_client-6.0.3.dev0-py2.py3-none-any.whl

**Comment by fschulze** [comment]
I released devpi-client 6.0.3.

**Comment by henryborchers** [comment]
Just an FYI. I've started moving to the latest version of devpi client and I'm no longer having this issue so far. Thanks again for your work.
